src/main/java/org/spongepowered/api/text/TextBuilder.java

```diff
@@ -0,0 +1,1124 @@
+     * in the chat.
+     *
+     * @param hoverAction The new hover action for the text
+     * @return This text builder
+     * @see Text#getHoverAction()
+     */
+    public TextBuilder onHover(@Nullable HoverAction<?> hoverAction) {
+        this.hoverAction = hoverAction;
+        return this;
+    }
+
+    /**
+     * Returns the current {@link ShiftClickAction} of this builder.
+     *
+     * @return The current shift click action or {@link Optional#absent()} if
+     *         none
+     * @see Text#getShiftClickAction()
+     */
+    public final Optional<ShiftClickAction<?>> getShiftClickAction() {
+        return Optional.<ShiftClickAction<?>>fromNullable(this.shiftClickAction);
+    }
+
+    /**
+     * Sets the {@link ShiftClickAction} that will be executed if the text is
+     * shift-clicked in the chat.
+     *
+     * @param shiftClickAction The new shift click action for the text
+     * @return This text builder
+     * @see Text#getShiftClickAction()
+     */
+    public TextBuilder onShiftClick(@Nullable ShiftClickAction<?> shiftClickAction) {
+        this.shiftClickAction = shiftClickAction;
+        return this;
+    }
+
+    /**
+     * Returns a view of the current children of this builder.
+     *
+     * <p>The returned list is unmodifiable, but not immutable. It will change
+     * if new children get added through this builder.</p>
+     *
+     * @return An unmodifiable list of the current children
+     * @see Text#getChildren()
+     */
+    public final List<Text> getChildren() {
+        return Collections.unmodifiableList(this.children);
+    }
+
+    /**
+     * Appends the specified {@link Text}s to the end of this text.
+     *
+     * @param children The texts to append
+     * @return This text builder
+     * @see Text#getChildren()
+     */
+    public TextBuilder append(Text... children) {
+        Collections.addAll(this.children, checkNotNull(children, "children"));
+        return this;
+    }
+
+    /**
+     * Appends the specified {@link Text}s to the end of this text.
+     *
+     * @param children The texts to append
+     * @return This text builder
+     * @see Text#getChildren()
+     */
+    public TextBuilder append(Iterable<? extends Text> children) {
+        Iterables.addAll(this.children, checkNotNull(children, "children"));
+        return this;
+    }
+
+    private static <T> Collection<T> asCollection(Iterable<T> iterable) {
+        return (Collection<T>) iterable;
+    }
+
+    /**
+     * Inserts the specified {@link Text}s at the given position of this builder.
+     *
+     * @param pos The position to insert the texts to
+     * @param children The texts to insert
+     * @return This text builder
+     * @throws IndexOutOfBoundsException If the position is out of bounds
+     * @see Text#getChildren()
+     */
+    public TextBuilder insert(int pos, Text... children) {
+        for (Text child : checkNotNull(children, "children")) {
```

Comment:
https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s4.6.2-horizontal-whitespace
> **On both sides of any binary or ternary operator.** This also applies to the following "operator-like" symbols:
- the ampersand in a conjunctive type bound: <T extends Foo & Bar>
- the pipe for a catch block that handles multiple exceptions: catch (FooException | BarException e)
- **the colon (:) in an enhanced for ("foreach") statement**